c++/src/ColumnReader.cc

```diff
@@ -100,13 +100,16 @@ namespace orc {
           return;
         }
       }
+      rowBatch.hasNulls = false;
     } else if (incomingMask) {
       // If we don't have a notNull stream, copy the incomingMask
       rowBatch.hasNulls = true;
       memcpy(rowBatch.notNull.data(), incomingMask, numValues);
       return;
+    } else {
+      memset(rowBatch.notNull.data(), 1, numValues);
```

Comment:
The contract of `ColumnVectorBatch` is that we should not depend on values from `ColumnVectorBatch.notNull` buffer if `hasNulls == false`. Probably we should update the site to explicitly state this: https://orc.apache.org/docs/core-cpp.html

Reply:
Adding an extra `memset` here does not break the contract but introduces a little bit performance penalty for non-null values.

Reply:
The problem is caused only when user misuses ColumnVectorBatch.notNull array without consuming hasNulls flag. Need to check if the java side follows the same contract.